### DISCUSSION TITLE
.ssh/config: line 3: Bad configuration option: usekeychain

### DISCUSSION THREAD

**Discussion by slomeo** [community/community#144078] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi all, 

from my lxplus.cern.ch account I try to push a commit:
"git push my-cmssw my_commit"

but I have this message:
...ssh/config: line 3: Bad configuration option: usekeychain
../.ssh/config: terminating, 1 bad configuration options
fatal: Could not read from remote repository.

my config is:

Host github.com
  AddKeysToAgent yes
  UseKeychain yes
  IdentityFile ~/.ssh/id_rsa.pub

I perfomed ssh-add ~/.ssh/id_rsa and in my GitHub account is present the key...

Where is my mistake?
Thanks in advance.
Best Reagrds

Sergio Lo Meo

**Comment by dalvarez2596** [comment]
Hi @slomeo  I think this might work : 
<img width="657" alt="Screenshot 2024-11-09 at 10 13 41 AM" src="https://github.com/user-attachments/assets/7dec0493-9a42-4f33-9a85-8a846391501a">
Add this :
```
Host github.com
  IgnoreUnknown UseKeychain
```
For further info refer to this : 
https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent

Remember if you had more accounts or something and have a :
```
Host *
  IgnoreUnknown UseKeychain    <---- Added here
  AddKeysToAgent yes
  UseKeychain yes
  IdentityFile ~/.ssh/id_rsa.pub
Host github.com
  HostName github.com
  User git
  IdentityFile ~/.ssh/id_rsa.pub
```
If not just added under the github configuration :
```
 Host github.com
  IgnoreUnknown UseKeychain    <---- Added here
  AddKeysToAgent yes
  UseKeychain yes
  IdentityFile ~/.ssh/id_rsa.pub
```
Hope it works 😃

**Reply by slomeo** [reply]
Hi @dalvarez2596 , thank you but I have alredy tried to add "IgnoreUnknown UseKeychain" but when I push I see:

Permissions 0644 for '/afs/cern.ch/user/s/slomeo/.ssh/id_rsa.pub' are too open.
It is required that your private key files are NOT accessible by others.
This private key will be ignored.
Load key "/afs/cern.ch/user/s/slomeo/.ssh/id_rsa.pub": bad permissions
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

any other suggestions?
Best Regards

Sergio

**Reply by dalvarez2596** [reply]
Oh I see, 
Well probably you will need to change the permmisions on the SSH key, what I would suggest would be the following :
1 . Test your local key with the following command:  ssh -T git@github.com.
If didnt work Ill start with the config to generate a new key and do all the setup with the following guide :
[Generate a new ssh key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent) and override the current one.
If you are stuck and keep having issues I can help you through [discord](https://discord.gg/fSyuSaHB) ,so I can guide you step by step to setup your ssh, cause depending of the OS the config is different.
